### PR TITLE
fix(MPTv2): Fix pipeline triggers for v2 templated pipelines.

### PR DESCRIPTION
**Pull request by jtk54**
The listener for pipeline triggers was not planning templated pipelines and hence not triggering on pipeline completion.

### CODE REVIEW

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/spring/DependentPipelineExecutionListener.groovy [hunk 1]
```diff
@@ -16,32 +16,55 @@
...
 import groovy.transform.CompileDynamic
 import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
 import static com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType.PIPELINE
 
 @Slf4j
 @CompileDynamic
+@Component
```
Comment (ezimanyi) [inline orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/spring/DependentPipelineExecutionListener.groovy]: I think moving this from an explicit `@Bean` to a `@Component` removed the gating on `@ConditionalOnExpression('${front50.enabled:true}')`

Also this moves us from constructor injection (which is preferred) to value injection...even moving this to a component you could still just autowire the constructor.

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/spring/DependentPipelineExecutionListener.groovy [hunk 2]
```diff
@@ -51,8 +74,17 @@ class DependentPipelineExecutionListener implements ExecutionListener {
     }
 
     def status = convertStatus(execution)
+    def allPipelines = front50Service.getAllPipelines()
+    if (pipelinePreprocessors) {
+      // Resolve templated pipelines if enabled.
+      allPipelines = allPipelines.findAll { it.type == 'templatedPipeline'}
+      .collect { pipeline ->
+        V2Util.planPipeline(contextParameterProcessor, pipelinePreprocessors, pipeline)
```
Comment (ezimanyi) [inline orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/spring/DependentPipelineExecutionListener.groovy]: Do we need to gate this to only get called on v2 templates, or is it a no-op for v1?

Comment (jtk54) [reply orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/spring/DependentPipelineExecutionListener.groovy]: We should no-op for v1 to preserve behavior.

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/spring/DependentPipelineExecutionListener.groovy (hunk 1 above)
Comment (jtk54) [reply orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/spring/DependentPipelineExecutionListener.groovy]: Agreed on both, thanks for pointing that out.

**Comment by jtk54** [comment]
@spinnakerbot cherry-pick 1.13

**Comment by jtk54** [comment]
I'll cook up a port for 1.13.

**Comment by jtk54** [comment]
Actually I think this fix was also built on https://github.com/spinnaker/orca/pull/2777, so I need to pick that one first.

**Comment by jtk54** [comment]
@spinnakerbot cherry-pick 1.13

**Pull request merged**